src/components/stack-chart/Canvas.js

```diff
@@ -240,6 +246,18 @@ class StackChartCanvasImpl extends React.PureComponent<Props> {
     const timeAtEnd: Milliseconds =
       timeAtViewportEnd + TIMELINE_MARGIN_RIGHT * timePerCssPixel;
 
+    // Compute the range for the "same width" drawing as well.
+    const sameWidthsIndexAtStart = Math.max(
+      0,
+      bisectionRight(sameWidthsIndexToTimestampMap, timeAtViewportStart) - 1
+    );
+    const sameWidthsIndexAtEnd = Math.min(
+      sameWidthsIndexToTimestampMap.length - 1,
+      bisectionLeft(sameWidthsIndexToTimestampMap, timeAtViewportEnd)
```

Comment:
Ah, can you explain with a comment why you used `bisectionRight` and `bisectionLeft` respectively? It would be good for future reference.